code/modules/atmospherics/machinery/components/binary_devices/valve.dm

```diff
@@ -40,6 +40,10 @@ It's like a regular ol' straight pipe, but you can turn it on and off.
 		flick("[valve_type]valve_[on][!on]-[set_overlay_offset(piping_layer)]", src)
 	icon_state = "[valve_type]valve_[on ? "on" : "off"]-[set_overlay_offset(piping_layer)]"
 
+/obj/machinery/atmospherics/components/binary/valve/can_crawl_through()
+	. = ..()
+	return . && on // valves should block whatever is trying to go through them, regardless of power
```

Comment:
This one will respect being powered despite being a valve. Once closed, it is closed regardless of power